### PR TITLE
Qwen2-VL Bug / Incompatibility Fixes

### PR DESCRIPTION
**Pull request by tyler-romero**
## Summary
Two fixes:
1. https://github.com/linkedin/Liger-Kernel/pull/276/files created a backwards-incompatible change. So that when users use transformers < 4.47.0.dev0 they will see different behavior from liger (which mimics behavior of 4.47.0.dev0). By adding a conditional we can support the behavior of all versions of transformers that include qwen2_vl.
2. Recent versions of qwen2_vl from transformers make various qwen2_vl specific tokens mandatory to specify, so those are now specified to avoid AttributeErrors.

## Testing Done
Tested locally against both transformers 4.46.2 and transformers 4.47.0.dev0 (from git+https://github.com/huggingface/transformers.git@52ea4aa589324bae43dfb1b6db70335da7b68654)

- Hardware Type: RTX 4090
- [x] run `make test` to ensure correctness
- [x] run `make checkstyle` to ensure code style
- [x] run `make test-convergence` to ensure convergence

### CODE REVIEW

File: src/liger_kernel/transformers/model/qwen2_vl.py [hunk 1]
```diff
@@ -100,27 +101,53 @@ def lce_forward(
...
         if attention_mask is not None:
             attention_mask = attention_mask.to(inputs_embeds.device)
-    # The code is copied from https://github.com/huggingface/transformers/pull/33487
-    if position_ids is None and input_ids is not None:
-        position_ids, _ = self.get_rope_index(
-            input_ids, image_grid_thw, video_grid_thw, attention_mask
-        )
+
+    if version.parse(transformers_version) > version.parse("4.46.2"):
+        # NOTE: this bug fix for qwen2-vl is not applied until transformers 4.47.0
```
Comment (tyler-romero) [inline src/liger_kernel/transformers/model/qwen2_vl.py:142]: This is the main bug fix, fixing backwards compatibility

File: test/convergence/test_mini_models_with_logits.py [hunk 2]
```diff
@@ -331,8 +331,15 @@
...
             attention_dropout=0.0,
+            # bos and eos set to match the Mistral-7B tokenizer used to create the test dataset
+            # https://huggingface.co/mistralai/Mistral-7B-v0.1/blob/main/config.json
             bos_token_id=1,  # 151643
             eos_token_id=2,  # 151645
+            vision_start_token_id=32765,  # vocab_size - 5
+            vision_end_token_id=32766,  # vocab_size - 4
+            vision_token_id=32767,  # vocab_size - 3
+            image_token_id=32768,  # vocab_size - 2
+            video_token_id=32769,  # vocab_size - 1
```
Comment (tyler-romero) [inline test/convergence/test_mini_models_with_logits.py:342]: Complete set of Qwen2-VL token ids must be explicitly specified in recent transformers (this was not the case in transformers v4.45.0)

File: test/convergence/test_mini_models.py [hunk 3]
```diff
@@ -519,44 +525,43 @@ def run_mini_model(
...
-        #             reason="Qwen2-VL not available in this version of transformers",
-        #         ),
-        #     ],
-        # ),
+        pytest.param(  # qwen2_vl requires slightly larger tolerances to pass this test after bug fix to qwen2_vl in transformers v4.47.0
+            "mini_qwen2_vl",
+            32,
+            1e-4,
+            torch.float32,
+            8e-6,  # 1e-8,
```
Comment (tyler-romero) [inline test/convergence/test_mini_models.py:533]: @ByronHsu acceptable?

**Review by ByronHsu — APPROVED** [review]
THanks!!

**Pull request merged**